osu.Game/Screens/Edit/Editor.cs

```diff
@@ -849,6 +849,20 @@ private void exportBeatmap()
             new LegacyBeatmapExporter(storage).Export(Beatmap.Value.BeatmapSetInfo);
         }
 
+        private void deleteDifficulty()
+        {
+            dialogOverlay?.Push(new PromptForDifficultyDeleteDialog(confirmDifficultyDelete, () => { }));
+        }
+
+        private void confirmDifficultyDelete()
+        {
+            var current = playableBeatmap.BeatmapInfo;
+            if (current is null) return;
+
+            beatmapManager.Hide(current);
```

Comment:
Not sure about showing to the user that the difficulty will be "deleted" when it appears to be just hidden.

Reply:
Agreed that is doesn't read well but the summary of the method `Hide` is "Delete a beatmap difficulty."
The `Delete` method appears to remove the whole beatmap. 
Happy to be steered on this one :)

Reply:
Doh I think I messed up my filter when playing with delete, I'll take another crack.

Reply:
Actually sorry for the noise but currently the `Delete` operates on `BeatmapSetInfo` not `BeatmapInfo` (my bad for not reading)
Is an implementation required for  `BeatmapInfo`? I don't want to go fiddling with stuff that I shouldn't without some guidance :)

Reply:
I would wait for a second opinion but I personally think difficulties should be purged/soft-deleted in editor rather than hidden (which could become an annoyance if a user creates random diffs and want to delete them, especially when we have flow for showing hidden beatmaps in song select, or if the users uses the "Restore all hidden difficulties" button in settings).

Reply:
Potentially adding a 'Hide instead' option or something to the dialogue here if that's desired but you are making a lot of sense.

Reply:
Hiding is the terminology and methodology used because I don't want users to be able to delete beatmaps from a set. This will break things going forward (such as comparing a beatmap set with its online version – we don't want random files to be missing).

I'll have to have a think about how this should be handled in the editor, but probably a hard delete with ample confirmation is okay for now.

Reply:
Would you like me to have a crack at adding a hard delete for beatmaps to the manager then or hold off while you have a think?
Any more confirmation than the current implementations dialogue? (2 stage / are you really sure?)

Reply:
Go for it. It should be pretty simple.

Please make sure to add test coverage. And make sure the editor loads another difficulty from the same beatmap if available.